_episodes/06-debugging.md

```diff
@@ -0,0 +1,309 @@
+please open an issue on https://github.com/ESMValGroup/ESMValTool/issues and
+attach the run/recipe_*.yml and run/main_log_debug.txt files from the output directory.
+~~~
+What suggestions would you give the researcher for fixing the error?
+>
+>> ## Solution
+>>
+>> 1. inspect the ``main_log.txt``
+>> 2. check the user-config.yml to see if the correct directory for input data is introduced.
+>> 3. check the available data, regarding exp, mip, ensemble, start_year, and end_year
+>> 4. check the variable name in the ``diag_timeseries_temperature`` section in the recipe.
+> {: .solution}
+{: .challenge}
+
+> ## Check pre-processed data
+>
+> The setting ``save_intermediary_cubes`` in the configuration file can be used
+save the pre-processed data. More information about this setting can be found at
+[Configuration]({{ page.root }}{% link _episodes/03-configuration.md %})
+{: .callout}
+
+The result of the pre-processors is passed to the ``diagnostic_timeseries.py`` script,
+that is introduced in the recipe as:
+
+~~~YAML
+40        scripts:
+41          timeseries_diag:
+42            script: ocean/diagnostic_timeseries.py
+~~~
+
+The diagnostic scripts are located in the folder ``diag_scripts`` in
+the ESMValTool installation directory ``path_to_esmvaltool``.
+To find ``path_to_esmvaltool`` on your system,
+see [Installation]({{ page.root }}{% link _episodes/02-installation.md %}).
+
+let's see if we can change the script path as:
+
+~~~YAML
+40        scripts:
+41          timeseries_diag:
+42            script: diag_scripts/ocean/diagnostic_timeseries.py
+~~~
+
+~~~bash
+  esmvaltool -c user-config.yml recipe_example.yml
+~~~
+
+~~~
+esmvalcore._task.DiagnosticError: Cannot execute script 'diag_scripts/examples/diagnostic.py' (esmvaltool/diag_scripts/diag_scripts/examples/diagnostic.py): file does not exist.
+2020-06-29 20:39:31,669 UTC [53008] INFO    If you suspect this is a bug or need help, please open an issue on https://github.com/ESMValGroup/ESMValTool/issues and attach the run/recipe_*.yml and run/main_log_debug.txt files from the output directory.
+~~~
+{: .error}
+
+The script path should be relative to ``diag_scripts`` directory.
+It means that the script ``diagnostic_timeseries.py`` is located in ``path_to_esmvaltool/diag_scripts/ocean/``.
+Alternatively, the script path can be an absolute path.
+To examine this, we change the script path and run the recipe:
+
+~~~YAML
+40        scripts:
+41          timeseries_diag:
+42            script: path_to_esmvaltool/diag_scripts/ocean/diagnostic_timeseries.py
+~~~
+
+~~~bash
+  esmvaltool -c user-config.yml recipe_example.yml
+~~~
+
+> ## Available recipe and diagnostic scripts
+>
+> ESMValTool provides a broad suite of
+[recipes and diagnostic](https://docs.esmvaltool.org/en/latest/recipes/index.html)
+scripts for different disciplines like atmosphere, climate metrics, future projections,
+IPCC, land, ocean, ....
```

Comment:
Again, this sentence is not finished and feels a bit out of place.

Reply:
It is now revised.